x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/PyTorchModelIT.java

```diff
@@ -330,55 +321,44 @@ private int sumInferenceCountOnNodes(List<Map<String, Object>> nodes) {
     }
 
     private void putModelDefinition(String modelId) throws IOException {
-        Request request = new Request("PUT", "/" + MODEL_INDEX + "/_doc/trained_model_definition_doc-" + modelId + "-0");
+        Request request = new Request("PUT", "_ml/trained_models/" + modelId + "/definition/0");
         request.setJsonEntity("{  " +
-            "\"doc_type\": \"trained_model_definition_doc\"," +
-            "\"model_id\": \"" + modelId +"\"," +
-            "\"doc_num\": 0," +
-            "\"definition_length\":" + RAW_MODEL_SIZE + "," +
             "\"total_definition_length\":" + RAW_MODEL_SIZE + "," +
-            "\"compression_version\": 1," +
             "\"definition\": \""  + BASE_64_ENCODED_MODEL + "\"," +
-            "\"eos\": true" +
-            "}");
-        client().performRequest(request);
-    }
-
-    private void createModelStoreIndex() throws IOException {
-        Request request = new Request("PUT", "/" + MODEL_INDEX);
-        request.setJsonEntity("{  " +
-            "\"mappings\": {\n" +
-            "    \"properties\": {\n" +
-            "        \"doc_type\":    { \"type\": \"keyword\"  },\n" +
-            "        \"model_id\":    { \"type\": \"keyword\"  },\n" +
-            "        \"definition_length\":     { \"type\": \"long\"  },\n" +
-            "        \"total_definition_length\":     { \"type\": \"long\"  },\n" +
-            "        \"compression_version\":     { \"type\": \"long\"  },\n" +
-            "        \"definition\":     { \"type\": \"binary\"  },\n" +
-            "        \"eos\":      { \"type\": \"boolean\" },\n" +
-            "        \"task_type\":      { \"type\": \"keyword\" },\n" +
-            "        \"vocab\":      { \"type\": \"keyword\" },\n" +
-            "        \"with_special_tokens\":      { \"type\": \"boolean\" },\n" +
-            "        \"do_lower_case\":      { \"type\": \"boolean\" }\n" +
-            "      }\n" +
-            "    }" +
+            "\"total_parts\": 1" +
             "}");
         client().performRequest(request);
     }
 
-    private void putVocabulary(List<String> vocabulary) throws IOException {
+    private void putVocabulary(List<String> vocabulary, String modelId) throws IOException {
         List<String> vocabularyWithPad = new ArrayList<>();
         vocabularyWithPad.add(BertTokenizer.PAD_TOKEN);
         vocabularyWithPad.addAll(vocabulary);
         String quotedWords = vocabularyWithPad.stream().map(s -> "\"" + s + "\"").collect(Collectors.joining(","));
 
-        Request request = new Request("PUT", "/" + VOCAB_INDEX + "/_doc/test_vocab");
+        Request request = new Request(
```

Comment:
Do we need a dedicated PUT vocabulary API?

Reply:
@davidkyle yes, but this pr is long already. It will come later.